backend/src/dependencies/store.py

```diff
@@ -63,13 +60,140 @@ def install_dependencies(dependencies: List[DependencyInfo]):
                 dependencies_to_install.append(dependency)
         elif not version:
             dependencies_to_install.append(dependency)
+    return dependencies_to_install
+
+
+def install_dependencies_sync_impl(dependency_info_array: List[DependencyInfo]):
+    subprocess.check_call(
+        [
+            python_path,
+            "-m",
+            "pip",
+            "install",
+            *[
+                pin(dep_info["package_name"], dep_info["version"])
+                for dep_info in dependency_info_array
+            ],
+            "--disable-pip-version-check",
+            "--no-warn-script-location",
+        ]
+    )
+    for dep_info in dependency_info_array:
+        package_name = dep_info["package_name"]
+        version = dep_info["version"]
+        installed_packages[package_name] = version
+
+
+def install_dependencies_sync(
+    dependencies: List[DependencyInfo],
+):
+    dependencies_to_install = get_deps_to_install(dependencies)
+    if len(dependencies_to_install) > 0:
+        install_dependencies_sync_impl(dependencies_to_install)
```

Comment:
Not a fan of those `_impl`. I believe python has module-private functions (`_name` I believe), so please use those when you need private things.

But do we need to split the this into 2 at all? Why not just:

```py
def install_dependencies_sync(
    dependencies: List[DependencyInfo],
):
    dependencies= get_deps_to_install(dependencies)
    if len(dependencies) == 0:
        return
    # impl
```

Same for the async version.